cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/TopicReplicaDistributionGoal.java

```diff
@@ -79,16 +78,28 @@ public boolean isActionAcceptable(BalancingAction action, ClusterModel clusterMo
    */
   @Override
   public ActionAcceptance actionAcceptance(BalancingAction action, ClusterModel clusterModel) {
+    String sourceTopic = action.topic();
+    Broker sourceBroker = clusterModel.broker(action.sourceBrokerId());
+    Broker destinationBroker = clusterModel.broker(action.destinationBrokerId());
+    int numLocalReplicasOfSourceTopic = sourceBroker.replicasOfTopicInBroker(sourceTopic).size();
+    int numRemoteReplicasOfSourceTopic = destinationBroker.replicasOfTopicInBroker(sourceTopic).size();
+
     switch (action.balancingAction()) {
       case REPLICA_SWAP:
-        return ACCEPT;
+        String destinationTopic = action.destinationTopic();
+        if (sourceTopic.equals(destinationTopic)) {
+          return ACCEPT;
+        }
+
+        int numLocalReplicasOfDestinationTopic = sourceBroker.replicasOfTopicInBroker(destinationTopic).size();
+        int numRemoteReplicasOfDestinationTopic = destinationBroker.replicasOfTopicInBroker(destinationTopic).size();
+
+        return (numRemoteReplicasOfSourceTopic < numLocalReplicasOfSourceTopic
+                && numLocalReplicasOfDestinationTopic < numRemoteReplicasOfDestinationTopic) ? ACCEPT : REPLICA_REJECT;
```

Comment:
I fee like the criteria can be change to

(num_replica_of_topic1_on_src_broker - avg_replica_of_topic1)^2 +
(num_replica_of_topic1_on_dst_broker - avg_replica_of_topic1)^2 + 
(num_replica_of_topic2_on_src_broker - avg_replica_of_topic2)^2 +
(num_replica_of_topic2_on_dst_broker - avg_replica_of_topic2)^2

this is consistent with this goal, which is minimizing the sum of deviation of replica distribution over all topics. 

we calculate this value before and after action and if it is decreasing, we accept the action.

I notice ClusterModel does not directly have topic replica average value, and we can get from
num_of_partition_of_topic * replica_factor_of_topic / number_of_broker.